### PR TITLE
feat(npg): CHK 1831 retrieve bin from payment method service

### PR DESCRIPTION
**Pull request by Fernando-Granato**
#### List of Changes
Added in case of authorization request to be forwarded to NPG the logic for card data retrieval using payment-methods.

#### Motivation and Context
The change was made to manage the authorization flow for NPG.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/transactions/services/TransactionsService.java [hunk 1]
```diff
@@ -776,14 +793,13 @@ NewTransactionResponseDto.ClientIdEnum convertClientId(
                 ).orElseThrow(() -> new InvalidRequestException("Null value as input origin"));
     }
 
-    private String extractBinFromPan(RequestAuthorizationRequestDto requestAuthorizationRequestDto) {
+    private Mono<Optional<String>> extractBinFromPan(RequestAuthorizationRequestDto requestAuthorizationRequestDto) {
```
Comment (infantesimone) [inline src/main/java/it/pagopa/transactions/services/TransactionsService.java:796]: rename this method?

Comment (infantesimone) [reply src/main/java/it/pagopa/transactions/services/TransactionsService.java:796]: the point is that `bin` is not retrieved only from the `pan`, so better rename method name

Comment (Fernando-Granato) [reply src/main/java/it/pagopa/transactions/services/TransactionsService.java]: Fixed with d232dd40676e53de7e9bbbf5080c3e2f99d49f69!

**Review by infantesimone — APPROVED** [review]

**Comment by infantesimone** [comment]
to check integration test fails

**Pull request merged**